aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,584 @@
-.. include:: /../common/stub-topic.txt
+Migrating HTTP Modules to Middleware
+====================================
 
-|stub-icon| Migrating HTTP Modules to Middleware
-======================================================
+By `Matt Perdeck`_
 
-.. include:: /../common/stub-notice.txt
+`Modules and HTTP Handlers <https://msdn.microsoft.com/en-us/library/bb398986.aspx>`_
+are no longer available in ASP.NET 5. 
+This article shows how to migrate your modules and handlers to the new ASP.NET 5 :ref:`middleware <fundamentals-middleware>`.
+
+.. contents:: In this article:
+  :local:
+  :depth: 1
+
+Handlers and modules revisited
+------------------------------
+Before proceeding to ASP.NET 5 middleware, let's first recap the use of handlers and modules in ASP.NET 4 and below: 
```

Comment:
I'd like to avoid the term "ASP.NET 4" because of existing and potential future versions. I.e. what about ASP.NET 4.5, 4.6, 4.7?